src/routes/docs/tutorials/nextjs/step-4/+page.markdoc

```diff
@@ -0,0 +1,219 @@
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Name"
+                id="name"
+                type="text"
+                required={true}
+                onChange={(e) => setName(e.target.value)}
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Email</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Email"
+                id="email"
+                type="email"
+                required={true}
+                onChange={(e) => setEmail(e.target.value)}
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Password</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Password"
+                id="password"
+                type="password"
+                required={true}
+                onChange={(e) => setPassword(e.target.value)}
+              />
+            </div>
+          </div>
+
+          <div className="u-padding-block-16">
+            <button
+              type="submit"
+              disabled={!email || !password}
+              className="button"
+            >
+              Sign Up
+            </button>
+          </div>
+        </form>
+      </div>
+    </section>
+  );
+};
+
+export default SignUp;
+```
+
+Now let's add the signup function. We will be using Appwrite's `create` method to create a new user. We will also be using `react-hot-toast` to show toast messages.
+
+
+```js
+const signup = async (e) => {
+  e.preventDefault();
+  try {
+    await account.create(ID.unique(), email, password, name);
+    await account.createEmailSession(email, password);
+    const jwt = await account.createJWT();
+    const user = await account.get();
+    cookie.set("jwt", jwt.jwt);
+    cookie.set("userid", user.$id);
+    toast.success("Signed up successfully");
+    router.push("/");
+  } catch (error) {
+    toast.error(error.message);
+  }
+};
+```
+
+After the user is created, we are creating a session using `createEmailSession` method. Then we are creating a JWT token using `createJWT` method. We are storing the JWT token and user id in the cookie. Then we are redirecting the user to the home page.
+
+
+# Login page {% #login-page %}      
+
+In your project under the app directory, create a new file `src/app/(pages)/login/page.js`. Next.js will automatically create a route for this page `/login`. Let's create the login form.
+
+```js
+"use client";
+
+import Link from "next/link";
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import toast from "react-hot-toast";
+import { useCookie } from "next-cookie";
+
+import { account } from "@/config/appwrite";
+
+const Login = (props) => {
```

Comment:
This makes me think that we should have *some* form of typing, even if it's using JSDoc rather than TS for full compatibility.